edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Docker/DockerConfig.cs

```diff
@@ -86,9 +88,40 @@ internal static CreateContainerParameters GetCreateOptions(string createOptions)
         }
 
         internal static string ValidateAndGetImage(string image)
+        {
+            return ValidateAndGetImage(image, new EnvironmentWrapper());
+        }
+
+        internal static string ValidateAndGetImage(string image, IEnvironmentWrapper env)
         {
             image = Preconditions.CheckNonWhiteSpace(image, nameof(image)).Trim();
+
+            if (image[0] == '$')
+            {
+                Match matchHost = ImageUpstreamRegex.Match(image);
+                if (matchHost.Success
+                    && (matchHost.Groups["path"]?.Length > 0)
+                    && (env != null))
```

Comment:
>&& (env != null) [](start = 20, length = 16)

seems unnecessary for internal methods

Reply:
I kinda agree but it was part of a previous comment: 

philipktlin 6 days ago • 
 Member
check null for env argument

It is defensive coding, I understand it too.